server/bleep/src/repo/iterator/git.rs

```diff
@@ -0,0 +1,151 @@
+use super::*;
+
+use anyhow::Result;
+use gix::ThreadSafeRepository;
+use regex::RegexSet;
+use tracing::error;
+
+use std::{collections::HashMap, path::Path};
+
+pub enum BranchFilter {
+    All,
+    Select(RegexSet),
+}
+
+impl BranchFilter {
+    fn filter(&self, branch: &str) -> bool {
+        match self {
+            BranchFilter::All => true,
+            BranchFilter::Select(patterns) => patterns.is_match(branch),
+        }
+    }
+}
+
+impl Default for BranchFilter {
+    fn default() -> Self {
+        Self::All
```

Comment:
Does this mean that by default we'll index _all_ branches. Shouldn't we only index `HEAD` by default?